src/Synth/LFO.cpp

```diff
@@ -173,44 +182,41 @@ void LFO::releasekey()
     // and divide by current ramp factor. It will be multiplied during fading out.
     outStartValue *= (1.0f - rampOnRelease);
     // store current time
-    releaseTimestamp = lfopars_.time->time();
+    releaseTimestamp = lfopars.time->time();
     // calculate fade out duration in frames
-    fadeOutDuration = lfopars_.fadeout * lfopars_.time->framesPerSec();
+    fadeOutDuration = lfopars.fadeout * lfopars.time->framesPerSec();
     // set fadeout state
     lfo_state = lfo_state_type::fadingOut;
 }
-
 float LFO::lfoout()
 {
     //update internals
-    if ( ! lfopars_.time || lfopars_.last_update_timestamp == lfopars_.time->time())
+    if ( ! lfopars.time || lfopars.last_update_timestamp == lfopars.time->time())
     {
-        waveShape = lfopars_.PLFOtype;
-        int stretch = lfopars_.Pstretch;
-        if(stretch == 0)
-            stretch = 1;
+        updatePars();
+        switch(lfopars.fel) {
         
-        const float lfostretch = powf(basefreq_ / 440.0f, (stretch - 64.0f) / 63.0f);
-
-        float lfofreq = lfopars_.freq * lfostretch;
-        phaseInc = fabsf(lfofreq) * dt_;
-
-        switch(lfopars_.fel) {
             case consumer_location_type_t::amp:
-                lfointensity = lfopars_.Pintensity / 127.0f; // [0...1]
+                lfointensity = lfopars.Pintensity / 127.0f; // [0...1]
                 break;
             case consumer_location_type_t::filter:
-                lfointensity = lfopars_.Pintensity / 127.0f * 4.0f; // [0...4] octaves
+                lfointensity = lfopars.Pintensity / 127.0f * 4.0f; // [0...4] octaves
                 break; 
             case consumer_location_type_t::freq:
             case consumer_location_type_t::unspecified:
-                lfointensity = powf(2, lfopars_.Pintensity / 127.0f * 11.0f) - 1.0f; // [0...2047] cent
+                lfointensity = powf(2, lfopars.Pintensity / 127.0f * 11.0f) - 1.0f; // [0...2047] cent
                 break;
         }
     }
     
-    float out = baseOut(waveShape, phase);
-    // Apply intensity
+    // refresh freq if tempo has changed
+    if (lfopars.numerator && lfopars.denominator && tempo != time.tempo) {
+        tempo = time.tempo;
+        float lfofreq = float(tempo) * float(lfopars.denominator)/(240.0f * float(lfopars.numerator));
+        phaseInc = fabsf(lfofreq) * dt;
+    }
+
+    float out = baseOut(waveShape, fmod(phase + (lfopars.Pstartphase + 63.0f) / 127.0f, 1.0f));
```

Comment:
In the code below... doesn't this PR remove the `lfopars.Pstartphase` part from `phase`, which will make every use of `phase` below behave different? I'd expected that your would rather do `fmod(phase += ...`.

Reply:
with this change adding Pstartphase is moved from once at init to cyclic. That's needed to tune the phase of two playing LFOs agains each other, or to tune the LFO phase to a synced delay.
+= works only if the startphase is only added once.

Reply:
OK, agreed that the "+= works only if the startphase is only added once". However, `phase` is still a different value after this line than before the PR, so you change behavior in the rest of this function. Isn't this wrong? Instead, can't you do `const float phaseWithStartPhase = fmod ...` and then replace all further use of `phase` in this function by `phaseWithStartPhase`?

Reply:
you are right. at least for lines 221 and 292 the startphase has to be included.